core/lib/dal/src/blocks_dal.rs

```diff
@@ -2720,22 +2706,46 @@ impl BlocksDal<'_, '_> {
         .and_then(|row| row.sealed_at.map(|d| d.and_utc())))
     }
 
+    pub async fn set_l1_batch_number_for_pending_miniblocks(
+        &mut self,
+        l1_batch_number: L1BatchNumber,
+    ) -> DalResult<()> {
+        sqlx::query!(
+            r#"
+            UPDATE miniblocks
+            SET
+                l1_batch_number = $1
+            WHERE
+                l1_batch_number IS NULL
+            "#,
+            i64::from(l1_batch_number.0)
+        )
+        .instrument("set_l1_batch_number_for_pending_miniblocks")
+        .with_arg("l1_batch_number", &l1_batch_number)
+        .execute(self.storage)
+        .await?;
+        Ok(())
+    }
+
     pub async fn set_protocol_version_for_pending_l2_blocks(
         &mut self,
         id: ProtocolVersionId,
+        l1batch_number: L1BatchNumber,
```

Comment:
nit: name is a bit confusing now as you can call method with any l1 batch number. `set_protocol_version_for_l2_blocks_in_l1_batch` would be better. Also `l1batch_number` -> `l1_batch_number `